### DISCUSSION TITLE
multicast and broadcast

### DISCUSSION THREAD

**Discussion by lightyear67** [cesanta/mongoose#1652] [Ask questions, get answers] [answered]
- My goal is: I want to receive multicast or broadcast data
- My question is: How can I use mongoose to receive multicast or broadcast data?

# Environment

- mongoose version: _7.7_

**Comment by scaprile** [comment, marked as answer]
You have to correctly configure your socket interface to provide those frames and IP datagrams for you. It depends on your OS/environment.
For Linux/Win32, [check here](https://gist.github.com/hostilefork/f7cae3dc33e7416f2dd25a402857b6c6) (no relation to them).
On Linux, something like this will call your fn with MG_EV_READ on every multicast UDP from the SSDP/UPnP group:
```
int main(void) {
  struct mg_mgr mgr;
  static struct mg_connection *c;
  struct ip_mreq mreq;

  mg_mgr_init(&mgr);
  c = mg_listen(&mgr, "udp://0.0.0.0:1900", fn, NULL);
  mreq.imr_multiaddr.s_addr = inet_addr("239.255.255.250");
  mreq.imr_interface.s_addr = htonl(INADDR_ANY);
  if (setsockopt((int) c->fd, IPPROTO_IP, IP_ADD_MEMBERSHIP, (char *) &mreq,
                 sizeof(mreq)) < 0) {
    return 1;
  }
  while (true) mg_mgr_poll(&mgr, 200);
  mg_mgr_free(&mgr);
  return 0;
}
```

**Reply by lightyear67** [reply]
Thank you for your answer, it works well. helps me quite a lot.

**Reply by cpq** [reply]
@lightyear67 can you elaborate, what is your use case for the broadcast / multicast?
Maybe it could be integrated into the mainstream.
